### PR TITLE
test(sessionclaims): update tests

### PR DESCRIPTION
**Pull request by porcellus**
## Summary of change

- Fixed smaller issues
- Made the email param optional in user-facing email verification functions
- Updated tests to work with new claims
- Adding new tests may be done in a separate PR

## Related issues

-   

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [x] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [x] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [x] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [x] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [x] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [x] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

### CODE REVIEW

**Comment by porcellus** [comment]
I'll also review/remove/move test cases in a separate PR, here, I just fixed all tests.

File: lib/ts/recipe/emailverification/index.ts [hunk 1]
```diff
@@ -25,10 +25,25 @@ export default class Wrapper {
 
     static EmailVerificationClaim = EmailVerificationClaim;
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:27]: Why did you move the fetching email part into the index file? (i don't mind it, but any specific reason why?)

File: lib/ts/recipe/emailverification/index.ts [hunk 2]
```diff
@@ -40,26 +55,70 @@ export default class Wrapper {
...
     }
 
-    static async revokeEmailVerificationTokens(userId: string, email: string, userContext?: any) {
-        return await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.revokeEmailVerificationTokens({
+    static async revokeEmailVerificationTokens(userId: string, email?: string, userContext?: any) {
+        const recipeInstance = Recipe.getInstanceOrThrowError();
+
+        if (email === undefined) {
+            const emailInfo = await recipeInstance.getEmailForUserId(userId, userContext);
+            console.log(emailInfo);
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:85]: remove console log

File: lib/ts/recipe/emailverification/index.ts [hunk 3]
```diff
@@ -40,26 +55,70 @@ export default class Wrapper {
...
+            if (emailInfo.status === "OK") {
+                email = emailInfo.email;
+            } else if (emailInfo.status === "EMAIL_DOES_NOT_EXIST_ERROR") {
+                return {
+                    status: "EMAIL_ALREADY_VERIFIED",
+                };
+            } else if (emailInfo) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:92]:
```suggestion
            } else {
```

File: lib/ts/recipe/emailverification/index.ts [hunk 4]
```diff
@@ -40,26 +55,70 @@ export default class Wrapper {
...
+        if (email === undefined) {
+            const emailInfo = await recipeInstance.getEmailForUserId(userId, userContext);
+            if (emailInfo.status === "OK") {
+                email = emailInfo.email;
+            } else if (emailInfo.status === "EMAIL_DOES_NOT_EXIST_ERROR") {
+                // Here we are returning OK since that's how it used to work, but a later call to isVerified will still return true
+                return {
+                    status: "OK",
+                };
+            } else if (emailInfo) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:115]:
```suggestion
            } else {
```

File: lib/ts/recipe/emailverification/index.ts [hunk 5]
```diff
@@ -40,26 +55,70 @@ export default class Wrapper {
...
+        const recipeInstance = Recipe.getInstanceOrThrowError();
+        if (email === undefined) {
+            const emailInfo = await recipeInstance.getEmailForUserId(userId, userContext);
+            console.log(emailInfo);
+
+            if (emailInfo.status === "OK") {
+                email = emailInfo.email;
+            } else if (emailInfo.status === "EMAIL_DOES_NOT_EXIST_ERROR") {
+                return true;
+            } else if (emailInfo) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:68]:
```suggestion
            } else {
```

File: lib/ts/recipe/emailverification/index.ts [hunk 6]
```diff
@@ -25,10 +25,25 @@ export default class Wrapper {
...
+
+        if (email === undefined) {
+            const emailInfo = await recipeInstance.getEmailForUserId(userId, userContext);
+            if (emailInfo.status === "OK") {
+                email = emailInfo.email;
+            } else if (emailInfo.status === "EMAIL_DOES_NOT_EXIST_ERROR") {
+                return {
+                    status: "EMAIL_ALREADY_VERIFIED_ERROR",
+                };
+            } else if (emailInfo) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:39]:
```suggestion
            } else {
```

File: lib/ts/recipe/session/utils.ts [hunk 7]
```diff
@@ -330,7 +331,7 @@ export async function updateClaimsInPayloadIfNeeded(
         logDebugMessage("updateClaimsInPayloadIfNeeded checking shouldRefetch for " + validator.id);
         if ("claim" in validator && (await validator.shouldRefetch(newAccessTokenPayload, userContext))) {
             logDebugMessage("updateClaimsInPayloadIfNeeded refetching " + validator.id);
-            const value = await validator.claim.fetchValue(this.getUserId(), userContext);
+            const value = await validator.claim.fetchValue(userId, userContext);
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/utils.ts:334]: Hmm.. that's weird. Why didn't TS complain about this.getUserId() not being present earlier?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/session/utils.ts (hunk 7 above)
Comment (porcellus) [reply lib/ts/recipe/session/utils.ts:334]: I'm not a 100% sure :/ But this actually shows something: we rarely need this for email verification (and that we need to add a specific test)

File: lib/ts/recipe/emailverification/index.ts (hunk 1 above)
Comment (porcellus) [reply lib/ts/recipe/emailverification/index.ts:27]: I didn't really move it - the claim and the API implementations still do the fetching themselves. I made the email optional because this way it matches the auth recipe specific methods, so it'd be an easier update for people (and I didn't have to update that many tests)

Comment (rishabhpoddar) [reply lib/ts/recipe/emailverification/index.ts:27]: ok makes sense.

**Review by rishabhpoddar — APPROVED** [review]
